nav2_behavior_tree/plugins/decorator/truncate_path_controller.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Sarthak Mittal
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+
+#include "nav_msgs/msg/path.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "nav2_util/geometry_utils.hpp"
+#include "behaviortree_cpp_v3/decorator_node.h"
+
+#include "nav2_behavior_tree/plugins/decorator/truncate_path_controller.hpp"
+
+namespace nav2_behavior_tree
+{
+
+TruncatePath::TruncatePath(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::DecoratorNode(name, conf),
+  distance_(1.0)
+{
+  getInput("distance", distance_);
+}
+
+inline BT::NodeStatus TruncatePath::tick()
+{
+  setStatus(BT::NodeStatus::RUNNING);
+
+  nav_msgs::msg::Path input_path;
+  nav_msgs::msg::Path output_path;
+
+  getInput("input_path", input_path);
+
+  output_path.header = input_path.header;
+
+  geometry_msgs::msg::PoseStamped final_pose = input_path.poses.back();
+
+  double distance_to_goal;
+  do {
+    distance_to_goal = nav2_util::geometry_utils::euclidean_distance(
+      input_path.poses.front(), final_pose);
+
+    output_path.poses.push_back(input_path.poses.front());
+    input_path.poses.erase(input_path.poses.begin());
+  } while (distance_to_goal > distance_);
+
+  double final_angle = atan2(
```

Comment:
You should check for nans

Reply:
But how it could be NaN using atan2? (x, y) is supposed to be valid for both poses, aren't they?

Reply:
if both X and Y are 0, it can return INF, NAN